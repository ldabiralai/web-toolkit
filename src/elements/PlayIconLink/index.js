import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import { actionOneDarkBase, coreLightMinus1, coreLightBase } from '../../colors';
import { fontFamilies } from '../../typography';
import PlayIcon from '../PlayIcon';
import Link from '../Link';

export const DivPlayIconBg = styled.div`
  display: inline-block;
  vertical-align: top;
  background-color: ${props => props.bgColorIcon};
  height: 100%;
  width: 48px;
  text-align: center;
`;

export const DivTextBg = styled.div`
  display: inline-block;
  vertical-align: top;
  background-color: ${props => props.bgColorText};
  color: ${coreLightBase};
  height: 100%;
  padding: 0 16px;
  text-align: center;
  font-family: ${fontFamilies.interUi};
  font-size: 10px;
  letter-spacing: 1px;
  text-transform: uppercase;
`;

const StyledPlayIcon = styled(PlayIcon)`
  display: inline-block;
  width: 38%;
  vertical-align: middle;
`;

const PlayIconLink = styled(({ children, bgColorIcon, bgColorText, ...props }) => {
  const Wrapper = props.href ? Link : styled.div();
  return (
    <Wrapper {...props}>
      <DivPlayIconBg bgColorIcon={bgColorIcon}>
        <StyledPlayIcon isRounded={false} />
      </DivPlayIconBg>
      <DivTextBg bgColorText={bgColorText}>{children}</DivTextBg>
    </Wrapper>
  );
})`
  position: absolute;
  color: ${coreLightMinus1};
  white-space: nowrap;
  height: 48px;
  line-height: 48px;
`;

PlayIconLink.propTypes = {
  href: PropTypes.string,
  target: PropTypes.string,
  bgColorIcon: PropTypes.string,
  bgColorText: PropTypes.string,
};

PlayIconLink.defaultProps = {
  bgColorIcon: actionOneDarkBase,
  bgColorText: '#3c46dc',
};

export default PlayIconLink;
