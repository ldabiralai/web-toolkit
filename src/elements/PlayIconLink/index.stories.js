import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, color } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';
import { PlayIconLink } from '../..';

const iconsStories = storiesOf('Elements|PlayIconLink', module);

iconsStories.add(
  'default',
  withInfo()(() => (
    <PlayIconLink
      bgColorIcon={color('BgColorIcon', '#4652ff')}
      bgColorText={color('bgColorText', '#3c46dc')}
      href={text('href', 'http://eurosport.fr')}
      target={text('target', '')}
    >
      {text('Link Text', 'Replay ')}
    </PlayIconLink>
  ))
);
