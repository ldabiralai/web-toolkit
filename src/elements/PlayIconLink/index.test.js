import React from 'react';
import { shallow, mount } from 'enzyme';
import PlayIconLink, { DivPlayIconBg, DivTextBg } from '.';

describe('PlayIconLink', () => {
  it('should render icon with link and text passed as children', () => {
    const wrapper = shallow(
      <PlayIconLink href="google.com" target="_blank" title="link title">
        REPLAY
      </PlayIconLink>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render a without href', () => {
    const wrapper = mount(<PlayIconLink />);
    expect(wrapper.getDOMNode().tagName).toEqual('DIV');
  });

  it('should use background colors provided to component', () => {
    const wrapper = shallow(
      <PlayIconLink bgColorIcon="red" bgColorText="blue">
        REPLAY
      </PlayIconLink>
    );
    expect(
      wrapper
        .dive()
        .find(DivPlayIconBg)
        .prop('bgColorIcon')
    ).toEqual('red');
    expect(
      wrapper
        .dive()
        .find(DivTextBg)
        .prop('bgColorText')
    ).toEqual('blue');
  });
});
