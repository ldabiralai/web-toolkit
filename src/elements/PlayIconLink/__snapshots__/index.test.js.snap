// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayIconLink should render icon with link and text passed as children 1`] = `
<Component
  bgColorIcon="#4652ff"
  bgColorText="#3c46dc"
  className="css-qh23ty exb52wv3"
  href="google.com"
  target="_blank"
  title="link title"
>
  REPLAY
</Component>
`;
